src/components/MyAccount/Settings/AccountSettingsDisplayOptions.tsx

```diff
@@ -39,35 +40,57 @@ export const AccountSettingsForm = ({ patron }: { patron: Patron }) => {
         | Iterable<ReactNode>
       switch (setting.term) {
         case "Home library":
-          inputField = (
-            <Select
-              name={setting.field}
-              id="update-home-library-selector"
-              labelText="Update home library"
-              showLabel={false}
-            >
-              <option value={patron.homeLibrary.code}>
-                {patron.homeLibrary.name}
-              </option>
-            </Select>
-          )
+          {
+            const patronHomeLibrary = getLibraryByCode(patron.homeLibraryCode)
+            const sortedPickupLocations = [
+              patronHomeLibrary,
+              ...filteredPickupLocations.filter(
+                (loc) => loc.code.trim() !== patron.homeLibraryCode.trim()
+              ),
+            ]
+            inputField = (
+              <Select
+                name={setting.field}
+                id="update-home-library-selector"
+                labelText="Update home library"
+                showLabel={false}
+              >
+                {sortedPickupLocations.map((loc, i) => (
+                  <option key={`location-option-${i}`} value={loc.code}>
+                    {loc.name}
+                  </option>
```

Comment:
From the my last testing, this is the only feature that threw an error:

<img width="966" alt="Screenshot 2024-05-10 at 9 28 23 AM" src="https://github.com/NYPL/research-catalog/assets/1280564/367e37a5-9037-45ca-8edc-3875f744877a">

I updated this to be `loc?.code` and `loc?.name` and that kind of fixed it. I updated the home library and saved it and now I don't get this error anymore. I wonder if it's one specific library code that's the problem?

Reply:
To be clear, I added the `?.` optional chaining and then _removed_ it and then it works as expected for various selected libraries. So it might've been an issue at first but gone after I updated the library value.

I would still be a bit defensive just in case.

Reply:
the homeLibrary update situation is actually a separate ticket! I think if this is not working here it's ok. I made some changes to the types that haven't been merged into this branch yet.